address_capabilities_bitmask.go

```diff
@@ -0,0 +1,73 @@
+package iotago
+
+import "github.com/iotaledger/hive.go/serializer/v2"
+
+const (
+	canReceiveNativeTokensBitIndex = iota
+	canReceiveManaBitIndex
+	canReceiveOutputsWithTimelockUnlockConditionBitIndex
+	canReceiveOutputsWithExpirationUnlockConditionBitIndex
+	canReceiveOutputsWithStorageDepositReturnUnlockConditionBitIndex
+	canReceiveAccountOutputsBitIndex
+	canReceiveNFTOutputsBitIndex
+	canReceiveDelegationOutputsBitIndex
+)
+
+type AddressCapabilitiesBitMask []byte
+
+func (bm AddressCapabilitiesBitMask) hasBit(bit int) bool {
+	byteIndex := bit / 8
+	if len(bm) <= byteIndex {
+		return false
+	}
+	bitIndex := bit % 8
+	return bm[byteIndex]&(1<<bitIndex) > 0
+}
+
+func (bm AddressCapabilitiesBitMask) setBit(bit int) AddressCapabilitiesBitMask {
+	newBitmask := bm
```

Comment:
should we maybe check here that `bit` isn't negative?

Reply:
let me change it to uint